### PR TITLE
Clarify documentation for `versioning-strategy` option

### PR DESCRIPTION
**Pull request by jeffwidman**
We receive the occasional support request and there are also multiple issues filed against https://github.com/dependabot/dependabot-core/ asking about "I tried X strategy for Y ecosystem but it's not working".

So rewrite this sections of the docs to add some clarity:
1. Add table illustrating how the different strategies work for a given scenario (🎩 💁‍♂️  to @jonasfj from https://github.com/dependabot/dependabot-core/issues/4979#issue-1196967453)
2. Invert the table from "strategy -> supported ecosystems" to "ecosystem -> supported strategies" because our users arrive with questions about a given ecosystem, not a given strategy
3. Add additional ecosystems to the list
4. Shorten the example because we only need one, additional ones just contribute to the #walloftext feeling
5. Re-order everything lexicographically

Hopefully this will provide better documentation that we can point users to in response to questions.

Co-authored-by: Jonas Finnemann Jensen <jonasfj@google.com>

### Why:

Closes [issue link]

### Check off the following:

- [ ] I have reviewed my changes in staging (look for the "Automatically generated comment" and click the links in the "Preview" column to view your latest changes).
- [ ] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

### CODE REVIEW

File: content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md
Comment (jeffwidman) [inline content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md:743]: These TODOs need to be filled out before merge. I work on the Dependabot team, so I can drive that.

File: content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md
Comment (jeffwidman) [inline content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md:679]: I was a little confused by the old docs... they imply all these ecosystems support `auto` which implies these ecosystems also support `widen` / `increase`... but not all these ecosystems are listed under `widen` / `increase`... is that an omission or am I misunderstanding how `auto` works?

File: content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md
Comment (jeffwidman) [inline content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md:755]: Changing the interval was intentional... see https://github.com/github/docs/pull/20734/

File: content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md
Comment (jeffwidman) [inline content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md:727]: Previously the docs implied that the default strategy is `auto` for _all_ ecosystems... However, regardless of whether true today, that may not be true in the future, for example https://github.com/dependabot/dependabot-core/issues/4979 so I added the `Default` column

File: content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md
Comment (Nishnha) [inline content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md:729]: `increase` and `increase-if-necessary` are misnamed. I can't find any record of them in our codebase.

I think these were meant to be `bump_versions` and `bump_versions_if_necessary`, which we support in [our dry-runner](https://github.com/dependabot/dependabot-core/blob/e11608fd17da3275cefca5e22eb01d01eb62fe09/bin/dry-run.rb#L196-L197) and were added to the bundler ecosystem in [Dependabot v0.85.2](https://github.com/dependabot/dependabot-core/blob/e11608fd17da3275cefca5e22eb01d01eb62fe09/CHANGELOG_ARCHIVE_2017_TO_2018.md?plain=1#L96-L99)

Comment (jeffwidman) [reply content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md:729]: 🤦 

I'm surprised that the incorrect config values weren't blowing up...probably they were silently getting swallowed as no-ops. I'll update this.

Comment (deivid-rodriguez) [reply content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md:729]: Actually turns out the `increase-if-necessary` and `increase` terminology is correct. The `bump_versions` and `bump_versions_if_necessary` are just internal naming, but the proper values for the configuration are the ones documented 👍.

**Comment by jeffwidman** [comment]
Not stale, just ahven't gotten to it yet

**Review by Yahwehloves — COMMENTED** [review]
Made a couple of changes to the request

File: content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md
Comment (sanderr) [inline content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md:741]:
```suggestion
| `pip` | `auto`, `increase`, `increase-if-necessary`, `lockfile-only`, `widen` | `auto` |
```
`increase` is definitely supported here, and with #6632 merged I think `widen` can be added as well?
`increase-if-necessary` I've never used myself, but from the error message I got when I tried to use `widen` I think it's supported as well:
`The property '#/updates/0/versioning-strategy' value "widen" did not match one of the following values: lockfile-only, auto, increase, increase-if-necessary`

**Review by sanderr — COMMENTED** [review]
Nice, it's a lot more clear than the current docs on this in my opinion.

File: content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md
Comment (deivid-rodriguez) [reply content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md:741]: Yes, all those 4 are currently supported for Python projects, and I plan to work on adding `widen` soon.

I think you meant to link to https://github.com/dependabot/dependabot-core/pull/6632, which fixed the update logic to better support `widen`, but we still need some internal work to expose the option in the configuration file.

Comment (sanderr) [reply content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md:741]: I see, thanks for the correction. I updated the suggested change and the link in my comment.

**Comment by abdulapopoola** [comment]
@jeffwidman is this part of the doc drive for this quarter?

**Comment by steveoh** [comment]
These docs would be a huge improvement. Where's the finish line?

File: content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md
Comment (jeffwidman) [reply content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md:815]: The code was difficult to trace, so I finally restored to testing this by setting up a test repo and trying different options. Thankfully the `dependabot.yml` configuration validator CI check would not only flag wrong values, but also list the allowed values, so it wasn't quite as painful as it could have been.

File: content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md
Comment (jeffwidman) [reply content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md]: I both spelunked in the code and also created a test repository... what I've got are the currently supported values.

File: content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md
Comment (jeffwidman) [reply content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md]: Yeah, I just tried on a test repo, and `widen` isn't yet supported, or at least the config file validator chokes on it. Thanks to @deivid-rodriguez 's hard work, some of the building blocks are in place, but there's a little more to do to get it across the finish line.

**Comment by jeffwidman** [comment]
@cmwilson21 this is ready for review.

Reviewers--if you have questions, feel free to also ping me on the internal company slack.

**Comment by cmwilson21** [comment]
@jeffwidman Thanks for letting me know! I'll get it on the board ⚡

File: content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md
Comment (mchammer01) [inline content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md:814]: Can we please use markdown for this table?

File: content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md
Comment (mchammer01) [inline content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md:864]:
```suggestion
| Ecosystem | Supported versioning strategies | Default strategy | Additional notes |
```

File: content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md
Comment (mchammer01) [inline content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md:885]:
```suggestion
# Example configuration for customizing the manifest version strategy
```

File: content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md
Comment (mchammer01) [inline content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md:882]: Typo + following the correct syntax for a note box.

```suggestion
{% note %}

**Note:** `N/A` indicates that the package manager does not yet support configuring the `versioning-strategy` parameter. The strategy code is open source, so if you'd like a particular ecosystem to support a new strategy, you are always welcome to submit a pull request in https://github.com/dependabot/dependabot-core/.

{% endnote %}
```

File: content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md
Comment (mchammer01) [inline content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md:866]: We don't seem to really need the "Additional notes" column as it seems to be a repeat of the `Default strategy` column, unless I misunderstood its purpose. Shall we remove it?

**Review by mchammer01 — COMMENTED** [review]
@jeffwidman 👋🏻 - thanks for improving our docs 💖 
Added a few comments and suggestions for your consideration.
I also need to check the guidance on alignment for table headers and normal cells so I'll come back to you on that one.
Let me know if you have any questions 😃

File: content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md
Comment (mchammer01) [inline content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md:812]: Following our content model:
```suggestion
The following table shows and example of how `versioning-strategy` can be used.
```

File: content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md
Comment (mchammer01) [inline content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md:809]: `Typically` requires a comma when placed at the start of a sentence.
```suggestion
| `widen`| Widen the allowed version requirements to include both the new and old versions, when possible. Typically, this only increases the maximum allowed version requirement. |
```

**Comment by jeffwidman** [comment]
Thanks! Let me incorporate these changes and then squash down to keep commit history clean.

File: content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md
Comment (jeffwidman) [reply content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md:812]: 👍 but let me fix the typo of `"shows and example"` 😁

Comment (mchammer01) [reply content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md:812]: Argh, so sorry about that 🙈

File: content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md
Comment (jeffwidman) [reply content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md:803]: Haha, great catch! I meant to delete it as part of the point of this PR is that users typically want to index by ecosystem => strategy, rather than the other way around. Fixed!

File: content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md
Comment (jeffwidman) [reply content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md:814]: The problem I ran into was I couldn't find a clean way to do nested tables with markdown... see the original place I grabbed this table from here in this post: https://github.com/dependabot/dependabot-core/issues/4979#issue-1196967453

I poked around stackoverflow and most of the answer recommended either HTML or a custom markdown flavor/extension, which probably isn't supported here.

I do think it renders better as a nested table, so I'm not sure what to do here.

Can you handle formatting it according to docs guidance? I'm not sure if you'll still want to do left-aligned for a nested table or some variation, so just do whatever docs team prefers.

Comment (jeffwidman) [reply content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md:814]: I just realized I'd opened this from my personal fork of docs 🤦 which can be painful to deal with as a maintainer even though I've given maintainers edit access, so if it's too annoying feel free to leave a comment with your update (or just ```suggestion I suppose works quick!)

File: content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md
Comment (RajaRam6) [reply content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md]: > Typo + following the correct syntax for a note box.

File: content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md
Comment (mchammer01) [reply content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md:814]: I am going to merge this great work, and will do the table conversion internally, so that we have the updates in our public-facing docs as soon as possible for our customers.

**Review by mchammer01 — APPROVED** [review]
LGTM :shipit:

**Pull request merged**
